.github/workflows/generate-changelog.yml

```diff
@@ -0,0 +1,30 @@
+name: Generate CHANGELOG
+on:
+  pull_request:
+    types: [closed]
+  workflow_dispatch:
+jobs:
+  GenerateChangelog:
+    if: github.event.pull_request.merged || github.event_name == 'workflow_dispatch'
```

Comment:
knitpit: this is strictly correct and efficient because it's not run if a PR is closed but not merged.

alternatively if we prefer it to be simpler we can remove that if condition.  It'll run sometimes when it's not necessary (when a pr is closed without merge) but there is not problem as this action is idempotent.

Reply:
also similarly to example.yaml we can use something like:

  pull_request:
 ...
    paths:
      - .changelog/**

so it will run only if the PR has an associated changelo entry file

but as my previous comment, I don't think it's very important to optimize run of this GH action

Reply:
I would avoid removing the if just in case a PR is closed with a new file in .changelog directory which could cause issues in the changelog

Reply:
doesn't it run with latest master content?

Reply:
yes you are correct, my bad. Even if it's run in latest master branch, I think it's better to keep the if. It will avoid  unnecessary jobs being triggered and help to keep the action history cleaner

Reply:
ok, then you might also want to add "paths" so it's only run when there is a changelog file as most of PRs won't have a changelog file

Reply:
Done!